spec/message.abnf

```diff
@@ -9,7 +9,7 @@ selectors = match 1*([s] expression)
 variant = when 1*(s key) [s] pattern
 key = nmtoken / literal / "*"
 
-expression = "{" [s] (((literal / variable) [s annotation]) / annotation) [s] "}"
+expression = "{" [s] (((literal / unquoted / variable) [s annotation]) / annotation) [s] "}"
 annotation = (function *(s option)) / reserved
 
 option = name [s] "=" [s] (literal / nmtoken / variable)
```

Comment:
@aphillips The ABNF change you suggest in https://github.com/unicode-org/message-format-wg/pull/364#discussion_r1199057894 would make it ambiguous whether the `value` of `foo=42` would get parsed as an `nmtoken` or an `unquoted`, so I presume that this is what you meant:

```suggestion
option = name [s] "=" [s] value
value = literal / nmtoken / variable
```


Would applying this change address your remaining concerns about this PR?

Reply:
@eemeli Not... exactly? My request to add `unquoted` to `option` was to make clear to readers that the value side of the option can contain an unquoted literal (which is what `unquoted` is). I agree that `nmtoken` is a pure superset of `unquoted`. Maybe the way out of this would be to add a comment to the ABNF (and specification) about the relationship between `unquoted` and `nmtoken`?